homeassistant/components/compit/climate.py

```diff
@@ -0,0 +1,284 @@
+            return []
+        return [
+            item.description
+            for item in self.available_presets.details
+            if item is not None
+        ]
+
+    @cached_property
+    def fan_modes(self) -> list[str] | None:
+        """Return the available fan modes."""
+        if self.available_fan_modes is None or self.available_fan_modes.details is None:
+            return []
+        return [
+            item.description
+            for item in self.available_fan_modes.details
+            if item is not None
+        ]
+
+    @property
+    def preset_mode(self) -> str | None:
+        """Return the current preset mode."""
+        preset_mode = self.get_parameter_value(PARAM_PRESET_MODE)
+
+        if preset_mode is None:
+            return None
+        if self.available_presets is None or self.available_presets.details is None:
+            _LOGGER.warning("The available_presets is None")
+            return None
+
+        val = next(
+            (
+                item
+                for item in self.available_presets.details
+                if item is not None and item.state == preset_mode.value
+            ),
+            None,
+        )
+        if val is None:
+            _LOGGER.warning("Current preset mode not found in available presets")
+            return None
+        return val.description
+
+    @property
+    def fan_mode(self) -> str | None:
+        """Return the current fan mode."""
+        fan_mode = self.get_parameter_value(PARAM_FAN_MODE)
+        if fan_mode is None:
+            return None
+
+        if self.available_fan_modes is None or self.available_fan_modes.details is None:
+            _LOGGER.warning("The available_fan_modes is None")
+            return None
+
+        val = next(
+            (
+                item
+                for item in self.available_fan_modes.details
+                if item is not None and item.state == fan_mode.value
+            ),
+            None,
+        )
+        if val is None:
+            _LOGGER.warning("Current fan mode not found in available fan modes")
+            return None
+        return val.description
+
+    @property
+    def hvac_mode(self) -> HVACMode | None:
+        """Return the current HVAC mode."""
+        hvac_mode = self.get_parameter_value(PARAM_HVAC_MODE)
+        if hvac_mode:
+            return COMPIT_MODE_MAP.get(hvac_mode.value)
+        return None
+
+    async def async_set_temperature(self, **kwargs: Any) -> None:
+        """Set new target temperature."""
+        temp = kwargs.get(ATTR_TEMPERATURE)
+        if temp is None:
+            raise ValueError("Temperature argument missing")
+        await self.async_call_api(PARAM_SET_TARGET_TEMPERATURE, temp)
+
+    async def async_set_hvac_mode(self, hvac_mode: HVACMode) -> None:
+        """Set new target HVAC mode."""
+
+        if not (mode := HVAC_MODE_TO_COMPIT_MODE.get(hvac_mode)):
+            raise ValueError(f"Invalid hvac mode {hvac_mode}")
```

Comment:
raise ServiceValidationError instead if its an user error. But be aware that the service will never be called with a hvac mode outside of the list defined in hvac_modes